app/controllers/mentor_to_mentee_matchers_controller.rb

```diff
@@ -13,16 +15,35 @@ def match
 
     ActiveRecord::Base.transaction do
       number = MentorToMenteeMatcher.new.run
+      create_projects
+    end
+
+    respond_to do |format|
+      format.json { render json: {number: number.abs}, status: :ok }
+    end
+  end
+
+  def rematch
+    ActiveRecord::Base.transaction do
+      MentorToMenteeMatcher.new.rematch
+      create_projects
 
-      ApplicationMatch.all.each do |application|
-        ProjectSetup.new(mentor_application: application.mentor_application,
-                         mentee_application: application.mentee_application)
-                    .create
+      MenteeApplication.where(state: 5).each do |application|
```

Comment:
as state is enum you can call here `MenteeApplication.waiting_for_rematch` instead of `MenteeApplication.where(state: 5)`

Reply:
Ah, you're right! Thanks for catching that :)